### PR TITLE
MSG-522 add option to specify jar classifier for searching nexus

### PR DESCRIPTION
**Pull request by mdowell12**
This allows a person to specify the `classifier` option in their `mendel.yml` file, which tells mendel to look for a jar in nexus with the pattern: `{jar_name}-{version}-{classifier}.jar`.  This is helpful when maven is being a punk and will only deploy uber jars with a `-shaded` classifier tacked on to the name.

### CODE REVIEW

**Comment by mdowell12** [comment]
screen shot 2019-01-23 at 5 48 14 pm

^^ proof that the integration tests passed

File: mendel/core.py [hunk 1]
```diff
@@ -483,7 +486,12 @@ def _generate_nexus_url(self):
         nexus_url += '/'
         nexus_url += self.project_version
         nexus_url += '/'
-        nexus_url += '{0}-{1}.jar'.format(self._jar_name, self.project_version)
+        nexus_url += '{0}-{1}'.format(self._jar_name, self.project_version)
+
+        if self._classifier is not None:
```
Comment (ethankhall) [inline mendel/core.py]: IDK how pythonic you want to be but `if self._classifier` would do.

Comment (Ibracadabra05) [reply mendel/core.py]: I agree with @ethankhall since `None` evaluates to false already

Comment (mdowell12) [reply mendel/core.py]: No thanks friends: https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_none.html

```
mattdowell@Matt-Dowells-MacBook-Pro:~/github/mendel$ python -c "import this" | grep "Explicit"
Explicit is better than implicit.
```
;) ;)

**Pull request merged**